### PR TITLE
DROOLS-3396 : CRLF in ACTION cell of xlsx spreadsheet is not treated as is

### PR DESCRIPTION
**Pull request by Rikkola**
https://issues.jboss.org/browse/DROOLS-3396

The error could be reproduced with any language, but kept the original test file as it is just in case it someday picks up some special character bugs.

@jomarko @manstis You two might be the best ones to review. Thank you.

### CODE REVIEW

File: drools-decisiontables/src/main/java/org/drools/decisiontable/parser/RhsBuilder.java [hunk 1]
```diff
@@ -89,10 +87,12 @@ public void clearValues() {
 
     public String getResult() {
         StringBuffer buf = new StringBuffer();
-        for ( Iterator<String> iter = this.values.iterator(); iter.hasNext(); ) {
-            buf.append( iter.next() );
+        for (Iterator<String> iter = this.values.iterator(); iter.hasNext(); ) {
+            String next = iter.next();
+            String replace = next.replace("\n", "\\n");
```
Comment (Rikkola) [inline drools-decisiontables/src/main/java/org/drools/decisiontable/parser/RhsBuilder.java:92]: Only thing that actually changed was I added this line in. Everything else is code formatting.

**Comment by Rikkola** [comment]
"org.drools.compiler.integrationtests.CepEspTest.testEventExpirationInSlidingWindow[KieBase type=STREAM_IDENTITY_MODEL_FLOW]"

jenkins please retest this

**Comment by Rikkola** [comment]
```
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.7.0:compile (default-compile) on project drools-examples: Compilation failure
/home/jenkins/workspace/KIE/master/pullrequest/drools-pullrequests/drools-examples/src/main/java/org/drools/examples/DroolsExamplesApp.java:[109,34] method execute in class org.drools.examples.helloworld.HelloWorldExample cannot be applied to given types;
  required: org.kie.api.KieServices,org.kie.api.runtime.KieContainer
  found: org.kie.api.runtime.KieContainer
  reason: actual and formal argument lists differ in length
```
jenkins please retest this

File: drools-decisiontables/src/test/java/org/drools/decisiontable/LineBreakXLSTest.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        KieSession ksession = kbase.newKieSession();
+
+        FactData fd = new FactData();
+        fd.set値(-1);
+        ksession.insert(fd);
+        ksession.fireAllRules();
+
+        ksession.dispose();
+
+        assertFalse(fd.getエラーメッセージ().contains(";"));
```
Comment (jomarko) [inline drools-decisiontables/src/test/java/org/drools/decisiontable/LineBreakXLSTest.java]: This assertion will always pass as the decision table is not setting semicolon in any rule. Please change to positive assertion:
```java
assertEquals(fd.getエラーメッセージ(), "値には0以上を指定してください。\n指定された値：")
```

**Review by jomarko — CHANGES_REQUESTED** [review]

**Comment by Rikkola** [comment]
Test updated.

**Review by jomarko — APPROVED** [review]
Thank you

**Comment by Rikkola** [comment]
@manstis hi, could you take a look please. One line change in the code + a new unit test was added.

**Review by manstis — APPROVED** [review]
Approving as it fixes the reported issue (however is it not equally likely that other control codes will cause similar problems; e.g. `\t`?)... anyway that can wait for another day (if it ever becomes an issue).

**Pull request merged**